### DISCUSSION TITLE
Reading commands from a file / improved Node.js repl integration

### DISCUSSION THREAD

**Discussion by jq170727** [hongaar/bandersnatch#443] [Ideas]
I've been using bandersnatch to interactively test libraries I'm developing.
I'm finding it would be helpful to have a consistent way in the repl to read and evaluate commands from a file.
This would allow me to easily reuse sequences of commands.

The [Node.js repl](https://nodejs.org/api/repl.html#commands-and-special-keys) provides **.save** and **.load** but I don't think they work in bandersnatch.
For example if I start my cli, run some commands and then enter **.save bar** 
I'll see a message `Session saved to: bar` but only an empty "bar" file will be created.

Similarly if I create a file "foo" containing some commands and run **.load foo** it looks like
the commands aren't parsed properly.  E.g.
```
> .load foo
init
signin user1 password1
signout

> (press enter)
Error: Unknown arguments: init
signin, user1, password1
signout
> 
```
Note my cli is suppressing the complete backtrace for the "Unknown arguments" error so you don't see it here.  
The commands in this example work fine if entered line at a time.  Also curious that the error report includes
commas in between the arguments of the second command.

From this I conclude some part of the Node.js repl is there but not fully integrated in bandersnatch.  
It would be nice if the Node.js .save/.load commands worked as they do in Node.js or if similar functionality 
was available in another form.

**Comment by hongaar** [comment]
I suspect this functionality is currently broken because of some enquirer interop workarounds, see e.g. https://github.com/hongaar/bandersnatch/blob/main/src/repl.ts#L28

Can you try to comment out lines 30 and 124 of this file, and see if it makes a difference?

**Reply by jq170727** [reply]
Thank you.  I tried your suggestion. 
After commenting out the corresponding lines in lib/esm/repl.js with pnpm patch 
(For reference here's the change I made: [bandersnatch@1.11.1.patch](https://github.com/hongaar/bandersnatch/files/11295847/bandersnatch%401.11.1.patch) )

I found the **.save** function now writes the commands to the file as expected.  🎉

But the **.load** behavior is unchanged - the saved commands seem to be parsed incorrectly.
